test/unit/linalg/tridiag.cpp

```diff
@@ -0,0 +1,112 @@
+#include <cfloat>
+#include <cstdlib>
+#include <gtest/gtest.h>
+#include <limits>
+#include <tuvx/linalg/linalg.h>
+#include <vector>
+
+using namespace tuvx::linalg;
+
+typedef trid_mat<double> trid_matd;
+typedef std::vector<double> vecd;
+
+typedef trid_mat<float> trid_matf;
+typedef std::vector<float> vecf;
+
+const int size = 10;
+const double tol_dp = 1e-14;
+const float tol_sp = 1e-6;
+
+// Demonstrate some basic assertions.
+TEST(TridiagSolveCPP, DoublePrecision) {
+  vecd x;
+  vecd b;
+  trid_matd A;
+
+  // initialize random matrix A and vector
+  fill_rand_mat(A, size);
+  fill_rand_vec(x, size);
+  b = dot(A, x);
+
+  // reconstruct x by tridiagonal solve
+  vecd x_approx = tridiag_solve(A, b);
+
+  // compute reconstruction error
+  double error = 0.0;
+  for (int i = 0; i < x_approx.size(); i++) {
+    error += std::pow(std::abs(x[i] - x_approx[i]), 2);
+  }
+  error = std::sqrt(error) / x_approx.size();
+
+  EXPECT_LE(error, tol_dp) << "DOUBLE EPSILON: "
+                           << std::numeric_limits<double>::epsilon()
+                           << std::endl;
+}
+
+TEST(TridiagSolveCPP, SinglePrecision) {
+  vecf x;
+  vecf b;
+  trid_matf A;
+
+  fill_rand_mat(A, size);
+  fill_rand_vec(x, size);
+
+  b = dot(A, x);
+
+  vecf x_approx = tridiag_solve(A, b);
+  float error = 0.0;
+  for (int i = 0; i < x_approx.size(); i++) {
+    error += std::pow(std::abs(x[i] - x_approx[i]), 2);
+  }
+  error = std::sqrt(error) / x_approx.size();
+
+  EXPECT_LE(error, tol_sp) << "FLOAT EPSILON: "
+                           << std::numeric_limits<float>::epsilon()
+                           << std::endl;
+}
+
+/*
+TEST(TridiagSolveCPP, SinglePrecisionAggregate) {
+
+  int n_trials = 10000;
+  int passed = 0;
+  int failed = 0;
+  double error = 0;
+
+  vecd x;
+  vecd b;
+  vecd x_approx;
+  trid_matd A;
+
+  vecd errors = vecd(n_trials);
+  for (int i = 0; i < n_trials; i++) {
+    fill_rand_mat(A, size);
+    fill_rand_vec(x, size);
+    b = dot(A, x);
+
+    x_approx = tridiag_solve(A, b);
+    error = 0.0;
+    for (int i = 0; i < x_approx.size(); i++) {
+      error += std::pow(std::abs(x[i] - x_approx[i]), 2);
+    }
+
+    error = std::sqrt(error) / x_approx.size();
+    errors[i] = error;
+
+    if (error < tol_dp) {
+      passed++;
+    } else {
+      failed++;
+    }
+  }
+
+  std::ofstream file;
+  file.open("../../../tool/numerics/errors_dbl.txt");
+  for (int i = 0; i < n_trials; i++) {
+    file << errors[i] << std::endl;
+  }
+  file.close();
+
+  EXPECT_EQ(failed, 0);
+}
+*/
```

Comment:
This seems like a useful test. However, we don't like to accept large blocks of commented code. Please either remove this test, uncomment the code block and add a description of what it's trying to test and turn it into a real test, or copy and paste this code into an issue describing what would need to be done to fix this test and delete the test.

Reply:
I have decided to remove it. There are always some edge cases where the precision errors cause the test to fail (for example an error of 1.1e-7 not meeting satisfying the error tolerance of 1e-7). I did reduce the tolerance to 1e-5 as suggested by Dr. Cao.